agent/pkg/providers/stats_provider.go

```diff
@@ -201,13 +251,24 @@ func convertAccumulativeStatsDictToArray(methodsPerProtocolAggregated map[string
 				EntriesCount:    entriesCount,
 				VolumeSizeBytes: volumeSizeBytes,
 			},
-			Color:   protocolToColor[protocolName],
+			Color:   getColor(protocolName),
 			Methods: methods,
 		})
 	}
 	return protocolsData
 }
 
+func getColor(protocolName string) string {
+	if protocolToColor == nil {
```

Comment:
can we do it in 'SetProtocolMap'? - it seems like 'SetProtocolMap' will be called once on init, and here we need locks for it to work properly